src/main/java/uk/gov/pay/api/model/CardDetails.java

```diff
@@ -36,6 +38,9 @@ public class CardDetails {
     @JsonProperty("card_type")
     private final String cardType;
 
+    @JsonProperty("wallet_type")
+    private String walletType;
```

Comment:
The only thing I'm slightly unsure about is that we use this class for both deserialisation of the response from connector/ledger and serialisation of the public API response. I wonder whether this is slightly confusing as someone might expect this field to be on the connector/ledger response.

I wonder whether it's clearer if we create a separate class to represent the public API response, and have a constructor method that accepts a CardDetails and the wallet type.

Reply:
I think it's confusing too, I've added a new class `CardDetailsFromResponse` to better distinguish between which DTOs are used for deserialisation/serialisation.